pkg/cloud/instance.go

```diff
@@ -161,13 +191,18 @@ func (c *client) GetOrCreateVMInstance(
 	if err != nil {
 		return err
 	}
+	diskOfferingID, err := c.ResolveDiskOffering(csMachine)
```

Comment:
Here we're setting the disk offering on the csMachine and returning it. Maybe we can drop the return and just set it? Most of the methods with Resolve in the name operate on the passed variable only.